### ISSUE TITLE
Translate & mirror regions aren't defined correctly

### ISSUE THREAD

**Issue by Alfalik** [twizmwazin/CardinalPGM#491]
+ Video proof: http://youtu.be/yIJEaWE0gzU
+ XML: http://pastebin.com/6c2mg147
+ Cardinal version 9f9f043

```xml
<!--  forward drop shafts  -->
	<cuboid name="forward-drop" min="0,0,0" max="7,50,7"/>
	<union name="forward-drops">
		<translate name="north-drop" region="forward-drop" offset="49,0,11"/>
		<translate name="east-drop" region="forward-drop" offset="76,0,38"/>
		<mirror name="west-drop" region="north-drop" normal="1,0,-1"/>
		<mirror name="south-drop" region="east-drop" normal="1,0,-1"/>
	</union>
```
Didn't find the other drop tube, I'll add a video/image if it appear

**Comment by iEli2tyree011** [comment]
Thanks for finding these. I'll have to figure out where the plugin isn't correctly defining each region.

**Comment by iEli2tyree011** [comment]
Working on these now

**Comment by Alfalik** [comment]
@iEli2tyree011 Translate regions are already fixed, as you can see in the video. Now you should check the mirror regions module.

+ Video: https://youtu.be/5ZGMGTVc4OY

**Comment by dentmaged** [comment]
@iEli2tyree011 Any updates?

**Comment by iEli2tyree011** [comment]
This is going to be a priority fix for 2.0, mirrored regions can play a big part in defining parts of a map and should be handled accordingly.

**Comment by Pablete1234** [comment]
@iEli2tyree011 do you know about: https://github.com/OvercastNetwork/Geometry? maybe that can help with regions

**Comment by iEli2tyree011** [comment]
@Pablete1234 Yeah, I think it would be easier to do the math on the spot, similar to translated regions. It would be done only once when modules load though. Still, it would make it much easier...that just seems like an extraneous dependency that we should only resort to if nothing else works.

**Comment by twizmwazin** [comment]
@Pablete1234 I could see that library being more useful as reference than as a dependency.

**Comment by Pablete1234** [comment]
@iEli2tyree011 @twizmwazin I was giving it to you so that you could reference code, cardinal doesn't like dependencies, it's better as it is, being a standalone plugin

**Comment by Alfalik** [comment]
What's the default location if ```origin="x,y,z"``` is null? I believe that's the problem. Look at this XML of NextGen that actually mirror the regions:

```xml
<union name="spawns">
    <cuboid name="red-spawn" min="-144,0,56" max="-173,9,80"/>
    <mirror name="blue-spawn" region="red-spawn" origin="-158.5,0,13.5" normal="0,0,-1"/>
</union>
```
And the BoomBox XML that does not mirror the region:
```xml
<union name="forward-drops">
    <translate name="north-drop" region="forward-drop" offset="49,0,11"/>
    <translate name="east-drop" region="forward-drop" offset="76,0,38"/>
    <mirror name="west-drop" region="north-drop" normal="1,0,-1"/>
    <mirror name="south-drop" region="east-drop" normal="1,0,-1"/>
</union>
```

**Comment by Pablete1234** [comment]
Fixed in 05d16cd
